### PR TITLE
Clarify VERILATOR_ROOT documentation (#4211)

### PR DESCRIPTION
**Pull request by Timmmm**
Clarify how `VERILATOR_ROOT` is used and embedded in the binary.

I also removed a duplicate installation option.

### CODE REVIEW

File: docs/guide/environment.rst [hunk 1]
```diff
@@ -91,9 +91,24 @@ associated programs.
...
-   Specifies the directory containing the distribution kit.  This is used
-   to find the executable, Perl library, and include files.  If not
-   specified, it will come from a default optionally specified at configure
-   time (before Verilator was compiled).  It should not be specified if
-   using a pre-compiled Verilator package as the hard-coded value should be
-   correct.  See :ref:`Installation`.
+   The ``VERILATOR_ROOT`` environment variable is used in several places:
+
+   * At ``./configure`` time: If set, it is embedded into the binary and used as
+     the default value for ``VERILATOR_ROOT`` if not specified at runtime.
```
Comment (wsnyder) [inline docs/guide/environment.rst:97]:
```suggestion
     the default value for :option:`VERILATOR_ROOT` if not specified at runtime.
```

(So it crossreferences.  Similar elsewhere.)

File: docs/guide/install.rst [hunk 2]
```diff
@@ -246,23 +247,7 @@ following:
    prepend-path PKG_CONFIG_PATH $install_root/share/pkgconfig
 
 
-3. Install into a Specific Path
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-You may eventually install Verilator into a specific installation prefix,
-as most GNU tools support:
```
Comment (wsnyder) [inline docs/guide/install.rst:253]: I think this does work, and is what distro authors need to use, can you comment on removal - was it confusing?

**Review by wsnyder — COMMENTED** [review]
I greatly appreciate these updates!  Some notes:

**Comment by wsnyder** [comment]
Merged manually, thanks.

File: docs/guide/install.rst (hunk 2 above)
Comment (Timmmm) [reply docs/guide/install.rst:253]: It does work but it's a duplicate of the previous section.

**Comment by Timmmm** [comment]
Sorry I totally missed your comments 2 weeks ago. Would you like me to make another PR to remove the duplicate install section?

**Pull request closed without merging**